### PR TITLE
Update product

### PR DESCRIPTION
**Pull request by at-hienpham**
## Description
Update product and fix create, update user
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
YES | NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Http/Controllers/Admin/ProductController.php [hunk 1]
```diff
@@ -31,4 +37,72 @@ public function show($id)
...
+     *
+     * @param App\Models\Product $product product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Product $product)
+    {
+        $stores = Store::pluck('name', 'id');
+        $categories = Category::pluck('name', 'id');
+        $images = Image::all();
```
Comment (at-vyle) [inline app/Http/Controllers/Admin/ProductController.php]: $images = $product->images;

File: resources/views/admin/pages/products/edit.blade.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                  <input type="text" name="name" class="form-control" value="{{ old('name', $product->name) }}" placeholder="{{ __('product.admin.create.enter_name') }}" />
+              </div>
+              <div class="form-group">
+                <label>{{ __('product.admin.show.images') }}</label>
+                <div class="form-line">
+                  <input type="file" name="path[]" class="form-control" multiple/>
+                </div>
+              </div>
+              @foreach ($images as $image)
+                @if($product->id == $image->product_id)
```
Comment (at-vyle) [inline resources/views/admin/pages/products/edit.blade.php]: không cần dùng if vì controller đã lấy hết ảnh của product này rồi

File: app/Http/Controllers/Admin/ProductController.php [hunk 3]
```diff
@@ -74,4 +76,48 @@ public function store(CreateProductRequest $request)
...
+     * @param \Illuminate\Http\Request $request request
+     * @param App\Models\Product       $product product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateProductRequest $request, Product $product)
+    {
+        try {
+            $product->update($request->all());
+            if (is_array(request()->file('path'))) {
```
Comment (at-quangtran) [inline app/Http/Controllers/Admin/ProductController.php]: vẫn dùng path?

File: resources/views/admin/pages/users/edit.blade.php [hunk 4]
```diff
@@ -15,18 +15,18 @@
...
                 <div class="form-group">
                   <label for="username">{{ __('user.admin.username') }}</label>
-                    <input type="text" name="username" class="form-control" value="{{ $user->username }}" readonly="true" placeholder="{{ __('user.admin.create.enter_username') }}" />
+                    <input type="text" name="username" class="form-control" value="{{ $user->username }}" readonly="true" disabled placeholder="{{ __('user.admin.create.enter_username') }}" />
                 </div>
                 <div class="form-group">
                   <label for="full_name">{{ __('user.admin.fullname') }}</label>
                   <div class="form-line">
-                    <input type="text" name="full_name" class="form-control" value="{{ $user->full_name }}" placeholder="{{ __('user.admin.create.enter_fullname') }}" />
+                    <input type="text" name="full_name" class="form-control" value="{{ old('full_name', $user->full_name) }}" placeholder="{{ __('user.admin.create.enter_fullname') }}" />
```
Comment (at-quangtran) [inline resources/views/admin/pages/users/edit.blade.php]: sửa chỗ này

File: resources/views/admin/pages/users/edit.blade.php [hunk 5]
```diff
@@ -15,18 +15,18 @@
...
                   </div>
                 </div>
                 <div class="form-group">
                   <label for="birthday">{{ __('user.admin.birthday') }}</label>
                   <div class="input-group">
-                    <input type="text" name="birthday" class="form-control" value="{{ $user->birthday }}" placeholder="{{ __('user.admin.create.enter_birthday') }}"/>
+                    <input type="text" name="birthday" class="form-control" value="{{ old('birthday', $user->birthday) }}" placeholder="{{ __('user.admin.create.enter_birthday') }}"/>
```
Comment (at-quangtran) [inline resources/views/admin/pages/users/edit.blade.php]: chỗ này nữa old('birthday')

**Pull request merged**